config/job.js

```diff
@@ -138,6 +152,7 @@ const SCHEMA_JOB = Joi.object()
         annotations: Annotations.annotations,
         description: SCHEMA_DESCRIPTION,
         environment: SCHEMA_ENVIRONMENT,
+        cache: SCHEMA_CACHE,
```

Comment:
I think it's weird the `cache` is part of `job` config but applies to everything in the pipeline?

The cache config itself looks like this: 
```
pipeline: ["node_modules/", "~/.sbt"]
event: ["target/generated-intermediate-resouce/*"]
job:
        - build: ["target/some-artifact.zip"]
        - publish-preview: ["target/some-artifact.zip"]
```

It doesn't really make sense to put this whole thing into a `job`, but based on this PR, it is valid.